### ISSUE TITLE
ReferenceError: document is not defined

### ISSUE THREAD

**Issue by ravi-pandit14** [mohitk05/react-insta-stories#51]
**Description** I am facing "_ReferenceError: document is not defined_" When import `import Stories from 'react-insta-stories';` in app.js React Native app.
I have installed library into the project successfully from `npm install --save react-insta-stories`

**What am i doing wrong?**

**Issue screenshot**:
![Screenshot 2019-12-30 at 5 49 27 PM](https://user-images.githubusercontent.com/30682376/71581892-8ef27b00-2b2d-11ea-9f66-a46ff730e6d3.png)

> My package.json
 ```
"react": "16.9.0",
"react-insta-stories": "^2.0.2",
"react-native": "0.61.5",
```

**Comment by AbdeenM** [comment]
Most probably this library is meant for reactjs web applications not react native. what a shame, just realized this after installing it and getting this error too

**Comment by mohitk05** [comment]
Yes, this is meant to be used in React application, but even for server rendered apps this error would show up because the final build contains `window` object. The reason is `style-loader` including window into the final build (https://github.com/webpack-contrib/style-loader/issues/109). One solution could be not importing styles until component is mounted. There are few solutions mentioned on this issue, we can implement one of these.

**Comment by iMerica** [comment]
Thanks for working on this project. I've forked it and switched to `isomorphic-style-loader` to render this component server-side in Next.js. 

I would love to submit a pull request, but the way you structured this project is very unusual. Normally, I'm able to fork a repo, make an enhancement on my fork and update my package.json to point to my personal Github fork. But that doesn't work because the root package.json doesn't have an entrypoint. (see #64)

Please consider moving `packages/react-native-stories` to a separate github repo and move `packages/react-insta-stories/*` to  the root of this repo.

**Comment by iMerica** [comment]
Upon further investigation, it appears that this package is not being built properly. There are no actual build dependencies in the [package.json](https://github.com/mohitk05/react-insta-stories/blob/master/packages/react-insta-stories/package.json).

> "dependencies": {}

You appear to be packaging the devDependencies into the final build that is being published to NPM. Nothing in this package requires `window` or `document`.

**Comment by xelahaippa** [comment]
In react applications with SSR and can be used by Lazy loading the component and disable SSR

    import dynamic from "next/dynamic";

    const DynamicComponentWithNoSSR = dynamic(() => import("../components/Stories"), {
      ssr: false,
    });

    export default () => <DynamicComponentWithNoSSR />;

**Comment by prafulrajhm** [comment]
Any solution for this?

I am facing same issue when I am using in Next.JS

**Comment by federicoromandemo** [comment]
> Any solution for this?
> 
> I am facing same issue when I am using in Next.JS

I am facing the same problem with Expo.

**Comment by prafulrajhm** [comment]
> > Any solution for this?
> > I am facing same issue when I am using in Next.JS
> 
> I am facing the same problem with Expo.

@xelahaippa solution worked for me, I created a component using `<DynamicComponentWithNoSSR />`

**Comment by KishanGujarati** [comment]
any update on this??

I am facing same issue when I am using in Next.JS

**Comment by cbasJS** [comment]
I had the same issue!!

I resolved using `@loadable/component` module.

```
// Packages
import loadable from '@loadable/component'

// Loadable components
const BeerStoryContent = loadable(() => import('../../components/BeerStoryContent'))

const ContainerComponent = () => {
  return (
   <Layout><BeerStoryContent /></Layout>
 )
}

export default ContainerComponent
```

**Comment by Manishbatra4** [comment]
@prafulrajhm can you show how I am facing same problem in nextjs

**Comment by cbasJS** [comment]
> @prafulrajhm can you show how I am facing same problem in nextjs

I really recommend you try using `@loadable/component`

I´m using Gatsby JS and this module is working perfectly for me =)

**Comment by ajmalafif** [comment]
> I had the same issue!!
> 
> I resolved using `@loadable/component` module.
> 
> ```
> // Packages
> import React from 'react'
> import loadable from '@loadable/component'
> 
> // Components
> import Layout from '../../components/Layout'
> 
> // Loadable components
> const BeerStoryContent = loadable(() => import('../../components/BeerStoryContent'))
> 
> const ContainerComponent = () => {
>   return (
>    <Layout><BeerStoryContent /></Layout>
>  )
> }
> 
> export default ContainerComponent
> ```

I can‘t seem to make it work with `@loadable/component` how did you made it work?

```
// src/components/LazyStories.js
import React from 'react'
import loadable from '@loadable/component'

const Story = loadable(() => import('./Story'))

const LazyStories = () => {
    return (
        <Story />
    )
}

export default LazyStories
```

```
// src/components/Story.js
import React from 'react'
import Stories from 'react-insta-stories'

export const Story = React.forwardRef(function Story(ref) {
  return (
    <Stories
      stories={ref} 
    />
  )
})
```

```
import { MDXProvider } from "@mdx-js/react"
import { MDXRenderer } from "gatsby-plugin-mdx"
import { LazyStories } from '../components/LazyStories'

const shortcodes = {LazyStories } 

<MDXProvider components={shortcodes}>
  <MDXRenderer>{mdx.body}</MDXRenderer>
</MDXProvider>
```

**Comment by mohitk05** [comment]
The module can now be loaded without any helper package. The fix has been added in v2.2.2!
